src/lib/debugger.ts

```diff
@@ -0,0 +1,59 @@
+import { reportRejection } from '@src/lib/trap'
```

Comment:
A global runtime class that will live for the runtime of the application that any part of the code base can push logs with `addLog`

This is used in many locations to track the flow of code and events that happen for the engine while the application is running. 

This is very useful when debugging the engine connection process.